packages/experimental/Switch/src/useSwitch.ts

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { useAsPressable, useKeyProps, useOnPressWithFocus, useViewCommandFocus } from '@fluentui-react-native/interactive-hooks';
+import { SwitchProps, SwitchInfo } from './Switch.types';
+import { LayoutAnimation } from 'react-native';
+import { InteractionEvent } from '@fluentui-react-native/interactive-hooks';
+
+const defaultAccessibilityActions = [{ name: 'Toggle' }];
+
+export const useSwitch = (props: SwitchProps): SwitchInfo => {
+  const defaultComponentRef = React.useRef(null);
+  const {
+    onChange,
+    checked,
+    defaultChecked,
+    label,
+    labelPosition,
+    componentRef = defaultComponentRef,
+    disabled,
+    accessibilityRole,
+    ...rest
+  } = props;
+
+  const isDisabled = !!disabled;
+  const initialCheckedState = !!(checked ?? defaultChecked);
+  const [checkedState, setCheckedState] = React.useState(initialCheckedState);
+  const [prevCheckedState, setPrevCheckedState] = React.useState(initialCheckedState);
+  const focusRef = isDisabled ? null : componentRef;
+
+  if (defaultChecked !== undefined && checked !== undefined) {
```

Comment:
We should gate this to only fire while in dev mode. console.logs cause memory leaks in react so we should avoid them in production.